src/uct/ib/rdmacm/rdmacm_cm.h

```diff
@@ -50,6 +56,36 @@ typedef struct uct_rdmacm_cm_config {
     double          timeout;
 } uct_rdmacm_cm_config_t;
 
+struct uct_rdmacm_cm_reserved_qpn_blk {
+    ucs_bitmap_t(256)       bmp;
+    uint32_t                first_qpn;
+    unsigned int            next_avail_qpn_offset;
+    ucs_list_link_t         entry;
+    struct mlx5dv_devx_obj  *obj;
+};
+
+struct uct_rdmacm_cm_reserved_qpn_context {
+     /* Avoid competed condition on the qpn resource for multi-threads */ 
+     pthread_mutex_t mutex;
+     ucs_list_link_t blk_list;
+     bool            support;
+     uint32_t        log_reserved_qpn_granularity;
+     uint64_t        general_obj_types_caps;
```

Comment:
why need to save this? after init phase we only case whether te reserved qpn feature is supported and what is the allocation block size

Reply:
@yosefe Fixed.